src/circuits/mimc/mimc.hpp

```diff
@@ -0,0 +1,52 @@
+// DISCLAIMER:
+// Content taken and adapted from:
+// https://github.com/HarryR/ethsnarks/blob/master/src/gadgets/mimc.hpp
+
+#ifndef __ZETH_MIMC_PERMUTATION_HPP__
+#define __ZETH_MIMC_PERMUTATION_HPP__
+
+#include "snarks_alias.hpp"
+#include "round.hpp"
+
+namespace libzeth  {
+ /*
+  * MiMCe7_permutation_gadget enforces correct computation of a MiMC permutation with exponent 7. It makes use of MiMCe7_round_gadget to enforce correct computation in each of the 91 rounds.
+  */
+template<typename FieldT>
+class MiMCe7_permutation_gadget : public libsnark::gadget<FieldT> {
+public:
+    std::vector<MiMCe7_round_gadget<FieldT>> round_gadgets;  // vector of MiMC round gadgets
+    std::vector<FieldT> round_constants;  //vector of round constants
+    static const int ROUNDS = 91; // nb of rounds suggested by the MiMC paper
+    const libsnark::pb_variable<FieldT> k;  // permutation key
+
+    // utility functions
+    // MiMC round gadgets initialization
+    void _setup_gadgets(
+        const libsnark::pb_variable<FieldT> in_x,
+        const libsnark::pb_variable<FieldT> in_k);
+
+    //Constants vector initialization
+    void _setup_sha3_constants();
```

Comment:
Why use a `_` prefix for the function name?

Reply:
inherited from the refactored code. Fixed.